### PR TITLE
Added a document of the deprecating process

### PR DESCRIPTION
**Pull request by AHB102**
Documented the general deprecation process in cri-o. 
Issue : Document configuration deprecation process #6959

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by AHB102** [comment]
I tried signing the commit with git commit -S, I will look into this why it isn't signing my commits

**Comment by haircommander** [comment]
I usually use `git commit -s` . now that it's commited, you can do `git commit --amend -s` and then do a force push

File: deprecating_process.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<p>Config options are the settings that allow us to control various aspects of the behaviour and functionality of CRIO. They allow us to customise how CRI-O operates within a kubernetes cluster.</p>
```
Comment (haircommander) [inline deprecating_process.md:1]: Can this have a title?

**Comment by haircommander** [comment]
this is a well written document, though the format is a bit tough to read. I think having more headers and titles would help.

**Comment by AHB102** [comment]
@haircommander I will make the necessary changes and commit them

**Comment by AHB102** [comment]
@haircommander I have added headings and titles and committed the changes by signing off with git commit --amend -s

File: deprecating_process.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# General Deprecating Process of CRI-O
+
+**What are config options?**
```
Comment (haircommander) [inline deprecating_process.md:3]: I would use `###` in place of the bold `**`

File: deprecating_process.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+1)General options: These control overarching settings like logging levels, storage locations, and state directories.
+
+2)Image options: These manage image storage, pulling, and default volumes.
+
+3)Network options: These configure container networking, including plugins and configuration directories.
+
+4)Runtime options: These govern the container runtime itself, such as the OCI runtime binary, pause command, and container monitor configuration.
+
+5)Security options: These enforce security measures like AppArmor and seccomp profiles.
```
Comment (haircommander) [inline deprecating_process.md:16]: personally I don't know if we need this section

File: deprecating_process.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Config options undergo deprecation only during major and minor version changes in CRI-O. Removals do not occur abruptly within patch releases, ensuring a smoother transition for users adapting to evolving configurations.
+
+A deprecated status is assigned to a configuration option for a minimum of one release cycle before it is actually removed. This gives users a warning and time to adjust their configurations to adapt to the upcoming changes.
+
+The deprecation is communicated through multiple channels, including updates in the documentation, messages to indicate deprecated config option in the CRI-O CLI help text and a log entry within CRI-O itself.
+
+Configuration options are typically removed if they are no longer considered necessary or if they have been replaced by alternatives, especially options present in the Kubelet.
+
+Several CRI-O config options have been replaced by Kubelet as the management of runtime containers has shifted towards the Kubelet.</p>
```
Comment (haircommander) [inline deprecating_process.md:28]: no need for </p> if below you use `###` instead of `**`

File: deprecating_process.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+**Examples of crio.conf options getting replaced by Kubelet:**
+
+1)Runtime Management was previously done with the “runtime\_path” config option which was replaced with the “--container-runtime” flag in Kubelet.
+
+2)Pod PIDs Limit was previously done with the “pids\_limit” config option which was replaced with the “--pods\_pids-limit” flag in Kubelet
+
```
Comment (haircommander) [inline deprecating_process.md:35]: can you replace the `1)` `2)` `3)` with `- `

**Comment by haircommander** [comment]
couple more notes, then LGTM, thanks @AHB102

**Comment by AHB102** [comment]
@haircommander I have made the recommended changes.

**Comment by haircommander** [comment]
LGTM, though the [linter](https://github.com/cri-o/cri-o/actions/runs/7501568789/job/20435638346?pr=7672) and [spellchecker](https://github.com/cri-o/cri-o/actions/runs/7501568789/job/20435638601?pr=7672)  aren't happy

**Comment by AHB102** [comment]
@AHB102 I am looking into it, will let you know.

**Comment by AHB102** [comment]
@haircommander I figured the linter part, but I really don't know what's wrong with spellcheck, so should I commit the linter part and see if it resolves the spellcheck part too ?

**Comment by haircommander** [comment]
sounds good @AHB102 !

**Comment by haircommander** [comment]
almost! https://github.com/cri-o/cri-o/actions/runs/7506839035/job/20439218081?pr=7672

**Comment by haircommander** [comment]
also I was wrong about spellcheck :facepalm: it was shellcheck and that's a known issue

File: deprecating_process.md [hunk 6]
```diff
@@ -1,21 +1,10 @@
...
-
-2)Image options: These manage image storage, pulling, and default volumes.
+## What are config options?
 
-3)Network options: These configure container networking, including plugins and configuration directories.
-
-4)Runtime options: These govern the container runtime itself, such as the OCI runtime binary, pause command, and container monitor configuration.
-
-5)Security options: These enforce security measures like AppArmor and seccomp profiles.
+Config options are the settings that allow us to control various aspects of the behaviour and functionality of CRIO. They allow us to customise how CRI-O operates within a kubernetes cluster.
```
Comment (TomSweeneyRedHat) [inline deprecating_process.md]: I'm not a fan of pronouns when writing documents. Something like this is more professional and more apparent, in my opinion.

```suggestion
Config options are the settings that can be used to control various aspects of the behavior and functionality of CRIO. They allow customization of CRI-O operations within a Kubernetes cluster.
```

At the very least, capitalize Kubernetes, and then consider using the American English version of "behavior" vs "behaviour". I'd also suggest changing "config" to "configuration" everywhere.

Comment (AHB102) [reply deprecating_process.md]: This is my first contribution to open source, still getting the hang of it. I will make it more formal.

**Comment by AHB102** [comment]
@haircommander ?

**Comment by haircommander** [comment]
/ok-to-test
/approve

LGTM, @cri-o/cri-o-maintainers PTAL

**Review by QiWang19 — COMMENTED** [review]
/lgtm

**Comment by QiWang19** [comment]
@AHB102 could you update the release note block in the first comment? It seems this will remove the label `do-not-merge/release-note-label-needed` and then this PR can merge.

**Comment by AHB102** [comment]
@TomSweeneyRedHat I'm happy to keep contributing! Your comments really improved the document, and I appreciate you taking the time to share them, even at the end of a long day. Thanks for your kind words, and I look forward to working with CRI-O more.

**Comment by AHB102** [comment]
@haircommander Couldn't have done it without you! Hopefully many more to come.

**Comment by haircommander** [comment]
well done thanks @AHB102 !

**Pull request merged**